src/file_storage.cpp

```diff
@@ -865,39 +867,39 @@ namespace libtorrent
 		return m_symlinks[fe.symlink_index];
 	}
 
-	time_t file_storage::mtime(internal_file_entry const& fe) const
+	std::time_t file_storage::mtime(internal_file_entry const& fe) const
 	{
-		int index = &fe - &m_files[0];
+		int const index = int(&fe - m_files.data());
```

Comment:
the this file and the ones below, if you would make ``index`` have type ``std::size_t`` you would save a bunch of casts I believe